src/amberscript/parser.cc

```diff
@@ -1005,6 +1005,11 @@ Result Parser::ParseRun() {
     if (!pipeline->IsGraphics())
       return Result("RUN command requires graphics pipeline");
 
+    if (pipeline->GetVertexBuffers().size() > 1)
+      return Result(
```

Comment:
Instead of doing this twice, can the code move into the Validate method in Pipeline? It's called at the end of both VkScript and AmberScript parsing.

Reply:
That's a good idea, however the way I see it, I need to pass in a pointer to commands to the validate function as an argument since pipeline object does not have access to script, right?

Reply:
Ah, crap. Good point, you don't know a drawRect has happened inside pipeline validation. The way you have it currently is good.